pkg/capabilities/triggers/mercury_trigger.go

```diff
@@ -26,6 +26,7 @@ var mercuryInfo = capabilities.MustNewCapabilityInfo(
 // This Trigger Service allows for the registration and deregistration of triggers. You can also send reports to the service.
 type MercuryTriggerService struct {
 	capabilities.CapabilityInfo
+	*capabilities.Validator
```

Comment:
Nit: if you make Validator a value type and add generics to it, then you can one line this addition into a fairly neat:

```
capabilities.Validator[mercury.FeedReport]
```

with no instantiation required.

Reply:
whoa, where can i read about this distinction / use case?